app/views/mixins.py

```diff
@@ -140,18 +140,38 @@ class FilterMixin(ListModelMixin):
     def get_queryset(self):
         emp_id_str = self.request.query_params.get("id")
         id = int(emp_id_str) if emp_id_str else None
+        date = self.request.query_params.get("date")
         start_date_str = self.request.query_params.get("start_date")
-        start_date = (
-            timezone.datetime.strptime(start_date_str, "%Y-%m-%d").date()
-            if start_date_str
-            else None
-        )
         end_date_str = self.request.query_params.get("end_date")
-        end_date = (
-            timezone.datetime.strptime(end_date_str, "%Y-%m-%d").date()
-            if end_date_str
-            else None
-        )
+
+        if not emp_id_str and not date and not start_date_str or not end_date_str:
+            print("ok")
+            return self.queryset.all()
+
+        if date == "weekly":
+            end_date_str = timezone.now().date()
+            start_date_str = end_date_str - timedelta(days=7)
```

Comment:
We are asked to calculate the last week days. If above ran on Tuesday of current week, Monday of last week will be missed. Try below code:
```suggestion
            some_day_last_week = timezone.now().date() - timedelta(days=7)
            monday_of_last_week = some_day_last_week - timedelta(days=(some_day_last_week.isocalendar()[2] - 1))
            monday_of_this_week = monday_of_last_week + timedelta(days=7)
```